reactive/hacluster.py

```diff
@@ -0,0 +1,41 @@
+from charms.reactive import when, when_not, clear_flag, set_flag
+from charms.reactive import endpoint_from_flag
+from charmhelpers.core import hookenv
+from charms.layer.kubernetes_common import get_ingress_address
+from charms.layer.hacluster import services
+
+
+@when('ha.connected', 'layer.hacluster.services_configured')
+@when_not('layer.hacluster.configured')
+def configure_hacluster():
+    """Configure HA resources in corosync"""
+    hacluster = endpoint_from_flag('ha.connected')
+    vips = hookenv.config('ha-cluster-vip').split()
+    dns_record = hookenv.config('ha-cluster-dns')
+    if vips and dns_record:
+        set_flag('layer-hacluster.dns_vip.invalid')
+        msg = "Unsupported configuration. " \
+              "ha-cluster-vip and ha-cluster-dns cannot both be set",
+        hookenv.log(msg)
+        return
+    else:
+        clear_flag('layer-hacluster.dns_vip.invalid')
+    if vips:
+        for vip in vips:
+            hacluster.add_vip(hookenv.application_name(), vip)
+    elif dns_record:
+        ip = get_ingress_address()
+        hacluster.add_dnsha(hookenv.application_name(), ip, dns_record,
+                            'public')
+
+    for name, service in services.items():
+        hacluster.add_init_service(name, service)
```

Comment:
What if the charm calls `remove_service_from_hacluster`? When is that service actually removed from the interface?

Reply:
Currently, it isn't and I don't know if this is expected. I do give that expectation with the name of the functions though. I'll figure this out.

Reply:
I've added https://github.com/openstack/charm-interface-hacluster/pull/1 to address deletions